src/amplitude.js

```diff
@@ -433,33 +552,45 @@ Amplitude.prototype.setUserId = function(userId) {
   try {
     this.options.userId = (userId !== undefined && userId !== null && ('' + userId)) || null;
     _saveCookieData(this);
-    // utils.log('set userId=' + userId);
   } catch (e) {
     utils.log(e);
   }
 };
 
-Amplitude.prototype.setOptOut = function(enable) {
-  if (!this._apiKeySet('setOptOut()')) {
+/**
+ * Sets whether to opt current user out of tracking.
+ * @public
+ * @param {boolean} enable - if true then no events will be logged or sent.
+ * @example: amplitude.setOptOut(true);
+ */
+Amplitude.prototype.setOptOut = function setOptOut(enable) {
+  if (!this._apiKeySet('setOptOut()') || !utils.validateInput(enable, 'enable', 'boolean')) {
```

Comment:
Have we always required API key to be set before any of these functions can be called, i.e. before multiple apps?